cli/utils.go

```diff
@@ -0,0 +1,209 @@
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/spf13/viper"
+
+	"github.com/prometheus/alertmanager/types"
+)
+
+const (
+	RESOLVE_EXACT = iota
+	RESOLVE_FUZZY = iota
+)
+
+func GetAlertmanagerURL() (*url.URL, error) {
+	u, err := url.ParseRequestURI(viper.GetString("alertmanager.url"))
+	if err != nil {
+		return nil, errors.New("Invalid alertmanager url")
+	}
+	return u, nil
+}
+
+// Parse a list of labels (cli arguments)
+func parseMatchers(labels []string, resolve int) (types.Matchers, error) {
```

Comment:
It's my hope to https://github.com/prometheus/alertmanager/pull/633 merged soon, so then the tool could accept a promql label matcher string (`{foo="bar", baz=~"qu.*"}`) and pass that on to the api.

this would be changing your api, however, so it's worth a discussion, and looping in @fabxc to see what he thinks. It looks like this might be expanding into generating a swagger api. Let's drop the discussion down to the main flow of the PR.